src/MicrosoftTeams.ts

```diff
@@ -795,6 +836,33 @@ namespace microsoftTeams {
             }
         }
 
+        /*
+        * Validates that the state param is a valid url with host as outlook.office.com and client_type correctly set to Outlook Desktop
+        * @param uri - the url to validate
+        */
+        function isValidState(state: string): boolean {
+            let link = document.createElement("a");
+            link.href = state;
+            if (link.host && link.host !== window.location.host && link.host === "outlook.office.com" && link.search.indexOf("client_type=Win32_Outlook") > -1) {
```

Comment:
This code seems very fragile and risky. Our SDK already has a concept of a hostClientType which tells us what kind of client we're being hosted in and that's sent down from the parent and not inferred ad-hoc from data fed to us by the 3rd party. Could we do something similar here where we introduce a win32outlook or win32desktop client type and the SDK decides on how to handle the notifySuccess/notifyFailure calls based on that?

Reply:
When we redirect the connector's page to the authentication url, there is no parent page reference to call the initialize() method which sets the hostClientType. 
We can add more checks here if required to ensure the 3rd party has not tempered with the value of the state param and we don't redirect to a malicious url.